torchao/dtypes/nf4tensor.py

```diff
@@ -120,6 +362,50 @@ def copy_(func, *args, **kwargs):
     return original.copy_(same_meta_nf4)
 
 
+@implements(
+    [
+        aten.is_pinned.default,
+    ]
+)
+def nf4_is_pinned(aten_op, args, kwargs=None):
+    nf4tensor = args[0]
+    return (
+        aten_op(nf4tensor.quantized_scalers, *(args[1:]), **kwargs) and
+        aten_op(nf4tensor.quantization_factor, *(args[1:]), **kwargs) and
+        aten_op(nf4tensor.quantized_data, *(args[1:]), **kwargs)
+    )
+
+
+@implements(
+    [
+        aten._pin_memory.default,
+    ]
+)
+def nf4_pin_memory(aten_op, args, kwargs=None):
+    nf4_tensor = args[0]
+    quantized_scalers = aten_op(nf4_tensor.quantized_scalers, *(args[1:]), **kwargs)
+    quantization_factor = aten_op(nf4_tensor.quantization_factor, *(args[1:]), **kwargs)
+    quantized_data = aten_op(nf4_tensor.quantized_data, *(args[1:]), **kwargs)
+    return NF4Tensor(
+        SubclassTensorArgs(
```

Comment:
nit: I seem to see this pattern a lot where we construct `SubclassTensorArgs` directly from an existing `nf4_tensor`. Perhaps, consider making this into a helper to avoid the duplication.

Reply:
haha. not nit at all. added util function to keep the code dry: `NF4Tensor(*construct_nf4_args(nf4tensor, updated_attrs))`